packages/grafana-data/src/vector/ArrayVector.ts

```diff
@@ -1,14 +1,13 @@
-import { MutableVector } from '../types/vector';
-
 import { FunctionalVector } from './FunctionalVector';
 
 /**
  * @public
  */
-export class ArrayVector<T = any> extends FunctionalVector<T> implements MutableVector<T> {
+export class ArrayVector<T = any> extends FunctionalVector<T> {
   buffer: T[];
 
-  constructor(buffer?: T[]) {
+  // This constructor needs to accept any otherwise the value takes over the definition
+  constructor(buffer?: any[]) {
```

Comment:
Somehow removing the nested complexity with MutableVector lets the typechecker be more strict... so the values passed into the buffer define the Vector<T> even if it was originally `any`

This change keeps the functional typing the same when used with `Field<T>` and constants defined -- mostly in the tests 🤷   See the `ArrayVector.test.ts` to see the typing that used to work and why this change is required

Reply:
> So the values passed into the buffer define the Vector
Don't understand this. Don't we want this?

Reply:
If someone wants any can't they just use `ArrayVector<any>` ?

Reply:
think I understand now :)